### ISSUE TITLE
Crash when closing the IM window right after a reply

### ISSUE THREAD

**Issue by GoogleCodeExporter** [marig345/prpltwtr#38]
```
twitter_api.c, around like 350:

static void twitter_api_set_statuses_success_cb(TwitterRequestor *r, xmlnode 
*node, gpointer _ctx)
{
    TwitterMultiMessageContext *ctx = _ctx;
    gboolean last = FALSE;

    if (++ctx->statuses_index >= ctx->statuses->len)
    {
        last = TRUE;
        //TODO: verify this doesn't leak
        g_array_free(ctx->statuses, TRUE);
        g_free(ctx);
    }
    if (ctx->success_func)
        ctx->success_func(r->account, node, last, ctx->user_data);
    }

    purple_debug_info(TWITTER_PROTOCOL_ID, "last is %d\n",last);
    if (last)
        return;


ctx->success_func shouldn't be called after g_free(ctx). Either add a return 
there, or wrap the call to success_func in an else [it doesn't matter, since if 
last == true, we return anyway]
```

Original issue reported on code.google.com by `mikeage` on 24 Jan 2011 at 1:07

**Comment by GoogleCodeExporter** [comment]
```
Same problem in twitter_api_send_dms_success_cb()

Same fix. 
```

Original comment by `mikeage` on 25 Jan 2011 at 2:59

**Comment by GoogleCodeExporter** [comment]
```
Here's a patched DLL [only change was in prpltwtr.dll, but I'm also uploading a 
matching gtkprpltwtr.dll]. This DLL also includes my old RT and reply-to-all 
patch, as well as the other stability fixes I've made. 
```

Original comment by `mikeage` on 25 Jan 2011 at 3:28

Attachments:
- [prpltwtr.dll](https://storage.googleapis.com/google-code-attachments/prpltwtr/issue-38/comment-2/prpltwtr.dll)
- [gtkprpltwtr.dll](https://storage.googleapis.com/google-code-attachments/prpltwtr/issue-38/comment-2/gtkprpltwtr.dll)

**Comment by GoogleCodeExporter** [comment]
```
Fixed in 0.6.0
```

Original comment by `mikeage` on 14 Feb 2011 at 6:45
- Changed state: **Verified**
